### DISCUSSION TITLE
Scalability and activity?

### DISCUSSION THREAD

**Discussion by garyrob** [elimintz/justpy#227] [Q&A] [answered]
Hi, as an experienced Python developer with minimal experience of the JavaScript world, I'm interested in using justpy for an upcoming web site project.

It could have thousands of simultaneous users, and could be a very long-term project. So I have a couple questions:

1) Based on the technologies justpy is based on, I'd tend to assume it can handle thousands of simultaneous users without becoming really slow, even with just one instance on a fast machine. These users would be interacting with it intermittently—reading some text, clicking a button to rate something, etc. (Not super-interactive like a game, at all.) Say, up to 10,000. Do you think that that assumption is likely to be correct?

1a) Have you tried running it under PyPy?

2) The most recent github check-in appears to be 5 months ago. Is this project going to be actively maintained and improved in the future?

**Comment by elimintz** [comment, marked as answer]
Hi,

Thanks for considering JustPy

To answer your questions:
1) One instance on a fast machine would be sufficient for the use case you describe.
1a) I haven't tried running under PyPy.

2) Yes, the project is actively maintained and is being constantly improved as I am using it myself. I hope to release a new version this month (there will be no breaking changes, mostly minor bug fixes and a new html parser) .

**Reply by garyrob** [reply]
OK, thanks for your answer! :)
